opentelemetry-aws/src/lib.rs

```diff
@@ -216,9 +216,9 @@ pub mod trace {
     #[derive(Clone, Debug, PartialEq)]
     struct XrayTraceId(String);
 
-    impl Into<Result<TraceId, ()>> for XrayTraceId {
-        fn into(self) -> Result<TraceId, ()> {
-            let parts: Vec<&str> = self.0.split_terminator('-').collect();
+    impl From<XrayTraceId> for Result<TraceId, ()> {
```

Comment:
I think `TryFrom` may be even more idiomatic here, i.e.:

```rust
use std::convert::TryFrom;

impl TryFrom<XrayTraceId> for TraceId {
    type Error = ();

    fn try_from(id: XrayTraceId) -> Result<Self, Self::Error> {
        // ...
    }
}
```

Probably out of scope for this PR, though.